core/src/main/scala/org/apache/spark/internal/config/History.scala

```diff
@@ -41,6 +41,15 @@ private[spark] object History {
     .timeConf(TimeUnit.SECONDS)
     .createWithDefaultString("10s")
 
+  val UPDATE_BATCHSIZE = ConfigBuilder("spark.history.fs.update.batchSize")
+    .doc("Specifies the batch size for updating new eventlog files. " +
+      "This controls each scan process to be completed within a reasonable time, and such " +
+      "prevent the initial scan from running too long and blocking new eventlog files to " +
+      "be scanned in time in large environments.")
+    .version("3.4.0")
+    .intConf
```

Comment:
We need a validation. IIUC, this should be positive to be valid. Do you have any other validity check logic?
```scala
.checkValue(s => s > 0, "The batch size should be positive")
```

Reply:
Added checkValue for a positive. A positive value should be enough.